### PR TITLE
OCPBUGS-44001: handle errors returned by createServicePrincipalWithCe…

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:

- handle errors returned by createServicePrincipalWithCertificate command
- use generated vnet and nsg resource group names

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [OCPBUGS-44001](https://issues.redhat.com/browse/OCPBUGS-44001)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/infra/azure/create.go [hunk 1]
```diff
@@ -165,14 +165,14 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 	} else {
 		// Create a resource group for network security group
 		nsgResourceGroupName = o.Name + "-nsg"
-		nsgRG, msg, err := createResourceGroup(ctx, o, azureCreds, nsgResourceGroupName, subscriptionID)
+		nsgResourceGroupName, msg, err = createResourceGroup(ctx, o, azureCreds, nsgResourceGroupName, subscriptionID)
```
Comment (bryan-cox) [inline cmd/infra/azure/create.go:168]: Is this not the same name as above?

File: cmd/infra/azure/create.go [hunk 2]
```diff
@@ -201,14 +201,14 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 	} else {
 		//create a resource group for virtual network
 		vnetResourceGroupName = o.Name + "-vnet"
-		vnetRG, msg, err := createResourceGroup(ctx, o, azureCreds, vnetResourceGroupName, subscriptionID)
+		vnetResourceGroupName, msg, err = createResourceGroup(ctx, o, azureCreds, vnetResourceGroupName, subscriptionID)
```
Comment (bryan-cox) [inline cmd/infra/azure/create.go:204]: Similar comment as above

File: cmd/infra/azure/create.go (hunk 1 above)
Comment (Patryk-Stefanski) [reply cmd/infra/azure/create.go:168]: nope when we create a resource group name we also have a unique identifier as a suffix

File: cmd/infra/azure/create.go (hunk 2 above)
Comment (Patryk-Stefanski) [reply cmd/infra/azure/create.go:204]: nope when we create a resource group name we also have a unique identifier as a suffix

**Pull request merged**